samples/native_training/test.sh

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash
+# Copyright 2022 The IREE Authors
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+# This script runs the steps laid out in the README for this sample. It is
+# intended for use on continuous integration servers and as a reference for
+# users, but can also be run manually.
+
+set -x
+set -e
+
+# Run under a virtual environment to isolate Python packages.
+#
+# For more information, see `build_tools/testing/run_python_notebook.sh`
+python3 -m venv .script.venv --system-site-packages --clear
+source .script.venv/bin/activate 2> /dev/null
+trap "deactivate 2> /dev/null" EXIT
+
+# Update pip within the venv (you'd think this wouldn't be needed, but it is).
+python3 -m pip install --quiet --upgrade pip
+
+# Install script requirements, reusing system versions if possible.
+python3 -m pip install --quiet \
+  -f https://iree-org.github.io/iree/pip-release-links.html iree-compiler
+python3 -m pip install --quiet \
+  -f https://llvm.github.io/torch-mlir/package-index/ torch-mlir
+python3 -m pip install --quiet \
+  git+https://github.com/iree-org/iree-torch.git
+
+# Update submodules in this repo.
+(cd $(git rev-parse --show-toplevel) && git submodule update --init)
+
+# Build the IREE runtime.
+(cd $(git rev-parse --show-toplevel) && cmake -GNinja -B /tmp/iree-build-runtime/ .)
+cmake --build /tmp/iree-build-runtime/ --target iree_runtime_unified
+
+# Build the example.
+cd $(git rev-parse --show-toplevel)/samples/native_training
+make
```

Comment:
https://buildkite.com/iree/iree-samples/builds/554#0184c9f7-5ff0-483f-979c-3091c6e8a49e

```
[2022-11-30T19:43:57Z] + make
--
  | [2022-11-30T19:43:57Z] /usr/bin/clang++-9 main.c -flto -Os -I../../runtime/src/ -L/tmp/iree-build-runtime//build_tools/third_party/flatcc/ -L/tmp/iree-build-runtime//runtime/src/iree/runtime/ -L/tmp/iree-build-runtime//third_party/cpuinfo/ -L/tmp/iree-build-runtime//third_party/cpuinfo/deps/clog -liree_runtime_unified -lflatcc_parsing -lcpuinfo -lclog -ldl -lpthread -o native-training
  | [2022-11-30T19:43:57Z] clang: warning: treating 'c' input as 'c++' when in C++ mode, this behavior is deprecated [-Wdeprecated]
  | [2022-11-30T19:43:58Z] /usr/bin/ld: error: Optimization level must be between 0 and 3
  | [2022-11-30T19:43:58Z] clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

you can also debug locally under Docker by copying this command (after configuring Docker on your machine): https://github.com/iree-org/iree/blob/693f45abe4e7e5fea8528c59f4d5e983abd04fdd/build_tools/buildkite/samples.yml#L16-L22

Reply:
Yep, that build is passing now: https://buildkite.com/iree/iree-samples/builds/556 (ignore the "Build host install", something else is happening there)